groupbytraceprocessor: TestPeriodicMetrics is unstable

See https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector/3878/workflows/16e17361-807e-43fb-995a-3e5a517bdef2/jobs/42647

```
2020-10-08T22:10:41.128Z	DEBUG	groupbytraceprocessor/event.go:93	recording current state of the queue	{"num-events": 0}
--- FAIL: TestPeriodicMetrics (0.05s)
    event_test.go:411: 
        	Error Trace:	event_test.go:411
        	            				event_test.go:360
        	Error:      	Not equal: 
        	            	expected: int(1)
        	            	actual  : float64(2)
        	Test:       	TestPeriodicMetrics
    event_test.go:411: 
        	Error Trace:	event_test.go:411
        	            				event_test.go:366
        	Error:      	Not equal: 
        	            	expected: int(0)
        	            	actual  : float64(1)
        	Test:       	TestPeriodicMetrics
```

--- a/processor/groupbytraceprocessor/event_test.go
+++ b/processor/groupbytraceprocessor/event_test.go
@@ -320,7 +320,12 @@ func TestEventShutdown(t *testing.T) {
 func TestPeriodicMetrics(t *testing.T) {
 	// prepare
 	views := MetricViews()
+
+	// ensure that we are starting with a clean state
+	view.Unregister(views...)
 	view.Register(views...)
+
+	// try to be nice with the next consumer (test)
 	defer view.Unregister(views...)
 
 	logger, err := zap.NewDevelopment()

